Starting with random values, we'll run 10 replicas

In [13]:
paste(as.character(runif(11)), collapse=", ")

[1] "0.133312059799209, 0.875859433785081, 0.518107586540282, 0.928697808180004, 0.77892731456086, 0.51010028924793, 0.920484015718102, 0.962446344085038, 0.527476714923978, 0.0429623052477837, 0.491534008877352"

Download samples

In [20]:
url = "https://adaphetnodes.shinyapps.io/design_of_experiments/_w_d48bee6e/session/29b169926b866dd268fc083e66234b46/download/downloadData?w=d48bee6e"
download.file(url, "data/doe_samples.csv", method="auto")

Import data

In [22]:
data = read.csv("data/doe_samples.csv")
summary(data)

                  Date         x1                x2               x3         
 2023-02-05-18:57:27:1   Min.   :0.01029   Min.   :0.1878   Min.   :0.06419  
 2023-02-05-18:58:50:1   1st Qu.:0.18255   1st Qu.:0.3529   1st Qu.:0.25536  
 2023-02-05-18:59:06:1   Median :0.48436   Median :0.7946   Median :0.53834  
 2023-02-05-18:59:56:1   Mean   :0.47084   Mean   :0.6396   Mean   :0.50627  
 2023-02-05-19:00:08:1   3rd Qu.:0.70855   3rd Qu.:0.8704   3rd Qu.:0.75987  
 2023-02-05-19:01:23:1   Max.   :0.94512   Max.   :0.9397   Max.   :0.92943  
 (Other)            :4                                                       
       x4                x5               x6               x7        
 Min.   :0.09173   Min.   :0.1096   Min.   :0.1751   Min.   :0.2423  
 1st Qu.:0.34569   1st Qu.:0.3481   1st Qu.:0.3534   1st Qu.:0.5202  
 Median :0.69870   Median :0.5020   Median :0.4404   Median :0.6990  
 Mean   :0.60410   Mean   :0.4999   Mean   :0.5207   Mean   :0.6466  
 3rd Qu.:0.81114   3rd Qu.

Linear model

In [27]:
m0 = lm(y ~ x1+x2+x3+x4+x5+x6+x7+x8+x9+x10+x11, data)
summary(m0)


Call:
lm(formula = y ~ x1 + x2 + x3 + x4 + x5 + x6 + x7 + x8 + x9 + 
    x10 + x11, data = data)

Residuals:
ALL 10 residuals are 0: no residual degrees of freedom!

Coefficients: (2 not defined because of singularities)
            Estimate Std. Error t value Pr(>|t|)
(Intercept)    4.284         NA      NA       NA
x1             2.269         NA      NA       NA
x2           -18.004         NA      NA       NA
x3            14.027         NA      NA       NA
x4            13.817         NA      NA       NA
x5             6.310         NA      NA       NA
x6            -1.986         NA      NA       NA
x7             6.734         NA      NA       NA
x8            -5.846         NA      NA       NA
x9           -24.119         NA      NA       NA
x10               NA         NA      NA       NA
x11               NA         NA      NA       NA

Residual standard error: NaN on 0 degrees of freedom
Multiple R-squared:      1,	Adjusted R-squared:    NaN 
F-statistic:   NaN on 9 and 0 D

> In this first model, x10 and x11 didn't figure in the coefficients, so we better discard them. However, the model as is could not fit all the variables (p-value NA) so we should look for the best candidates.

Pearson Correlation

In [ ]:
library("")

In [31]:
cor.test(data$y,data$x1)
cor.test(data$y,data$x2)
cor.test(data$y,data$x3)
cor.test(data$y,data$x4)
cor.test(data$y,data$x5)
cor.test(data$y,data$x6)
cor.test(data$y,data$x7)
cor.test(data$y,data$x8)
cor.test(data$y,data$x9)


	Pearson's product-moment correlation

data:  data$y and data$x1
t = 0.28027, df = 8, p-value = 0.7864
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 -0.5661701  0.6856637
sample estimates:
       cor 
0.09860729 



	Pearson's product-moment correlation

data:  data$y and data$x2
t = 1.4189, df = 8, p-value = 0.1937
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 -0.2525272  0.8406786
sample estimates:
      cor 
0.4483925 



	Pearson's product-moment correlation

data:  data$y and data$x3
t = -0.73964, df = 8, p-value = 0.4807
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 -0.7613454  0.4479923
sample estimates:
      cor 
-0.252996 



	Pearson's product-moment correlation

data:  data$y and data$x4
t = -0.96069, df = 8, p-value = 0.3648
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 -0.7910524  0.3862254
sample estimates:
      cor 
-0.321609 



	Pearson's product-moment correlation

data:  data$y and data$x5
t = -0.32277, df = 8, p-value = 0.7551
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 -0.6934990  0.5559332
sample estimates:
       cor 
-0.1133793 



	Pearson's product-moment correlation

data:  data$y and data$x6
t = -0.4782, df = 8, p-value = 0.6453
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 -0.7206865  0.5172087
sample estimates:
       cor 
-0.1667047 



	Pearson's product-moment correlation

data:  data$y and data$x7
t = 0.53948, df = 8, p-value = 0.6042
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 -0.5014178  0.7307781
sample estimates:
      cor 
0.1873588 



	Pearson's product-moment correlation

data:  data$y and data$x8
t = -0.44407, df = 8, p-value = 0.6688
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 -0.7149122  0.5258804
sample estimates:
       cor 
-0.1551012 



	Pearson's product-moment correlation

data:  data$y and data$x9
t = -3.7401, df = 8, p-value = 0.005705
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 -0.9500972 -0.3374263
sample estimates:
       cor 
-0.7976004 


> The absolute correlation winner is x9, but we cannot rely on a single variable. Let's build a second model with the best 3 variables.

**x9, x2 and x4**

In [32]:
m1 = lm(y ~ x9 + x2 + x4, data)
summary(m1)


Call:
lm(formula = y ~ x9 + x2 + x4, data = data)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.69846 -0.30232 -0.00339  0.24918  0.75990 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)  
(Intercept)  2.34117    0.82175   2.849   0.0292 *
x9          -2.17063    0.89939  -2.413   0.0523 .
x2          -0.06158    0.79437  -0.078   0.9407  
x4          -0.47502    0.60346  -0.787   0.4611  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.5176 on 6 degrees of freedom
Multiple R-squared:  0.6774,	Adjusted R-squared:  0.5161 
F-statistic:   4.2 on 3 and 6 DF,  p-value: 0.06388


> Already better. Now we made a valid model. Yet, it is still not good, with a coefficient of determination of only 67.74%. Let's add more variables to see if the R² value increases. The choice criterium will still be the smallest correlation p-value. 

In [38]:
m2 = lm(y ~ x9 + x2 + x4 + x3 + x7 + x6 + x8 + x5, data)
summary(m2)


Call:
lm(formula = y ~ x9 + x2 + x4 + x3 + x7 + x6 + x8 + x5, data = data)

Residuals:
        1         2         3         4         5         6         7         8 
-0.234325  0.054792 -0.062532  0.108777  0.133882 -0.006281 -0.053401  0.067052 
        9        10 
 0.044492 -0.052455 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)
(Intercept)   2.8147     0.7370   3.819    0.163
x9           -5.6769     4.1136  -1.380    0.399
x2           -1.4460     3.6259  -0.399    0.758
x4            0.9620     2.7090   0.355    0.783
x3            1.8143     2.9142   0.623    0.645
x7           -0.5322     1.9605  -0.271    0.831
x6            0.1190     0.6912   0.172    0.891
x8           -1.2851     1.2891  -0.997    0.501
x5            2.5703     1.1961   2.149    0.277

Residual standard error: 0.322 on 1 degrees of freedom
Multiple R-squared:  0.9792,	Adjusted R-squared:  0.8127 
F-statistic: 5.883 on 8 and 1 DF,  p-value: 0.3091


> Chosing all variables but x1, x10 and x11 we got to a coefficient of determination of 97.92%! That will do.